### PR TITLE
Support keywords by enclosing column identifiers by backtick characters

### PR DESCRIPTION
**Pull request by danidelvalle**
The goal of this PR is to support reserved keywords as column names. As per [BigQuery documentation](https://cloud.google.com/bigquery/docs/reference/standard-sql/lexical#reserved_keywords), some reserved keywords cannot be used as identifiers unless enclosed by backtick (`) characters. 

For example, the below merge query attempting to update a table with two columns (`id` and `order`; being the second a reserved keyword):

```sql
MERGE `test_table`.`foo` USING 
( 
  SELECT * FROM 
    (SELECT ARRAY_AGG(x ORDER BY i DESC LIMIT 1)[OFFSET(0)] src
     FROM `test_table`.`foo_tmp` x 
     WHERE batchNumber=0 GROUP BY key.id, key.__dbz__physicalTableIdentifier)
)
ON `foo`.kafkaKey=src.key
WHEN MATCHED AND src.value IS NOT NULL THEN UPDATE SET id=src.value.id, order=src.value.order
WHEN MATCHED AND src.value IS NULL THEN DELETE
WHEN NOT MATCHED AND src.value IS NOT NULL THEN INSERT (kafkaKey, _PARTITIONTIME, id, order);
```

will fail with the following BigQuery error message:

```json
GET https://www.googleapis.com/bigquery/v2/projects/REDACTED/queries/REDACTED
{
  "code" : 400,
  "errors" : [ {
    "domain" : "global",
    "location" : "q",
    "locationType" : "parameter",
    "message" : "Syntax error: Unexpected keyword ORDER at [1:X]",
    "reason" : "invalidQuery"
  } ],
  "message" : "Syntax error: Unexpected keyword ORDER at [1:X]",
  "status" : "INVALID_ARGUMENT"
}
```

### CODE REVIEW

**Review by C0urante — COMMENTED** [review]
Thanks @danidelvalle. This looks like a good start but it seems like there are still some places where we need to add backtick syntax to ensure that field names are properly quoted. I could be off, though; if there's BigQuery documentation detailing exactly when we need to quote and when we don't, could you point me in that direction?

On a separate note, have you tested this against BigQuery? Not necessarily but running the connector in upsert/delete mode, but at least verifying that a query that the connector currently constructs and fails on (because of column names that match keywords) no longer fails with the changes proposed here?

File: kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/MergeQueries.java [hunk 1]
```diff
@@ -187,11 +187,11 @@ SELECT ARRAY_AGG(
       )
       ON `<destinationTable>`.<keyField>=src.key
       WHEN MATCHED AND src.value IS NOT NULL
-        THEN UPDATE SET <valueField>=src.value.<field>[, <valueField>=src.value.<field>...]
+        THEN UPDATE SET `<valueField>`=src.value.<field>[, `<valueField>`=src.value.<field>...]
```
[the remark this answers is not in the document]
Comment (danidelvalle) [reply kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/MergeQueries.java:190]: Thanks @C0urante for your feedback. 

I'm afraid that I'm not aware of any more detailed BigQuery documentation on this apart from the link I already shared. However, let me explain how I tested this:

- With the changes of this branch, I managed to pass all integration tests, being especially relevant `UpsertDeleteBigQuerySinkConnectorIT`.

- I have built a Confluent Hub Archive as per README, and deployed it in a real-world scenario a couple of weeks ago without issues so far.

- Manually in the BigQuery portal. It parses the query nearly in real-time while typing, even without executing it and highlights mistakes. Addressing your comments about when to quote and when don't, I have inferred from this manual testing that the right-hand side doesn't require it - may be because of the dotted syntax (e.g.: `src.value.<field>`) does not produce any ambiguity.

Elaborating a little more in the 3rd, and playing with the SQL example I provided in the first comment, the BQ Query Editor highlights an error for not quoting the word "order":

error

Once fixed, it jumps to the other mistake at line 11. Notice that for BQ line 9 is OK without quoting the right-hand part of the assignment:

error2

To sum up, I think that additional quoting will not break anything but might be redundant. In other clauses, as in `GROUP BY`, `ON`, etc. the column identifiers are always part of the dotted syntax and do not appear alone, which I think is the root cause of the issue. 

Thoughts?

File: kcbq-connector/src/test/java/com/wepay/kafka/connect/bigquery/MergeQueriesTest.java [hunk 2]
```diff
@@ -123,19 +123,20 @@ public void testUpsertQueryWithPartitionTime() {
...
-              + "f1, f2, f3, f4) "
+              + "`f1`, `f2`, `f3`, `f4`) "
             + "VALUES ("
               + "src.key, "
               + "CAST(CAST(DATE(src.partitionTime) AS DATE) AS TIMESTAMP), "
               + "src.value.f1, src.value.f2, src.value.f3, src.value.f4"
             + ");";
     String actualQuery = mergeQueries(true, true, false)
         .mergeFlushQuery(INTERMEDIATE_TABLE, DESTINATION_TABLE, BATCH_NUMBER);
+    System.out.println(actualQuery);
```
Comment (C0urante) [inline kcbq-connector/src/test/java/com/wepay/kafka/connect/bigquery/MergeQueriesTest.java:139]: Can we remove this part? The failure messages from `assertEquals` should include both the expected query and the actual query if there are any test failures.

**Review by C0urante — APPROVED** [review]
LGTM. Since this is a fix for a bug that affects versions 2.0.x onward, we should target the 2.0.x branch with this change.

There are a few ways to do that, but one way is to checkout the 2.0.x branch from this repo, cherry-pick the commits from this PR onto that branch, and then force-push the result to the `feature/enclose-identifiers-in-merges` branch. After that, you can change the target branch in the GitHub UI (click the `edit` button next to the PR title) from `master` to `2.0.x`.

If this is too much of a pain let me know and I can push the changes directly to your branch.

This will allow us to port the change automatically across the 2.0.x, 2.1.x, 2.2.x, and master branches, and then we can perform new releases on the 2.0.x, 2.1.x, and 2.2.x branches with this bug fix.

File: kcbq-connector/src/test/java/com/wepay/kafka/connect/bigquery/MergeQueriesTest.java (hunk 2 above)
Comment (danidelvalle) [reply kcbq-connector/src/test/java/com/wepay/kafka/connect/bigquery/MergeQueriesTest.java]: Sure, done!

**Comment by danidelvalle** [comment]
> There are a few ways to do that, but one way is to checkout the 2.0.x branch from this repo, cherry-pick the commits from this PR onto that branch, and then force-push the result to the feature/enclose-identifiers-in-merges branch. After that, you can change the target branch in the GitHub UI (click the edit button next to the PR title) from master to 2.0.x.

Makes sense, done! Let me know if there is anything else I should do.

**Comment by C0urante** [comment]
Looks great! Thanks.
